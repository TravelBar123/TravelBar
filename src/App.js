import { Main } from './styles.js';
import NavBar from './components/NavBar';
import { useState,useEffect } from 'react';
import Axios from 'axios';
import MainPage from './components/MainPage/index.jsx';
import AllDrinksPage from './components/AllDrinksPage/index.jsx';
import SelectionAlcoholPage from './components/SelectionAlcoholPage/index.jsx';
import SelectionNoAlcoholPage from './components/SelectionNoAlcoholPage/index.jsx';

function App() {
  const [data, setData] = useState()
  const [see, setSee] = useState('main')

  useEffect(() => {
    Axios.get('https://www.thecocktaildb.com/api/json/v1/1/filter.php?c=Cocktail')
    .then(response => {
      setData(response.data)
    })
  },[setData])

    return (
      <Main className="App">
        <NavBar title='TravelBar' view = { setSee } />
        {see === 'main'?
        (<MainPage data= { data } setSee = { setSee } />)
        : see === 'all'?
        (<AllDrinksPage data= { data } setSee = { setSee } />)
        : see === 'alcohol'?
        (<SelectionAlcoholPage setSee = { setSee } />)
        : see === 'noAlcohol'?
        (<SelectionNoAlcoholPage setSee = { setSee } />)
        : (<p>oops...something went wrong</p>)
        }
      </Main>
    )
}
export default App;