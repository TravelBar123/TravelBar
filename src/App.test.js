import React from 'react'
import {render, cleanup, fireEvent } from '@testing-library/react'
import App from './App'

afterEach(cleanup)

it('should take a snapshot from App', () => {
    const { asFragment } = render(<App />)    
    expect(asFragment(<App />)).toMatchSnapshot()
});

it('should be Selection of Cocktail', () => {
    const { getByTestId } = render(<App />); 
    expect(getByTestId('selection')).toHaveTextContent('Selection of Cocktail')
});

it('Selection all cocktails', () => {
    const { getByTestId } = render(<App />);     
    fireEvent.click(getByTestId('btn'))
    expect(getByTestId('selection')).toHaveTextContent('All Cocktails')
});

it('Selection Alcohol', () => {
    const { getByTestId } = render(<App />);     
    fireEvent.click(getByTestId('btnAlcohol'))
    expect(getByTestId('selection')).toHaveTextContent('Cocktails With alcohol')
});

it('Selection No Alcohol', () => {
    const { getByTestId } = render(<App />);     
    fireEvent.click(getByTestId('btnNOAlcohol'))
    expect(getByTestId('selection')).toHaveTextContent('Cocktails Without alcohol')
});
