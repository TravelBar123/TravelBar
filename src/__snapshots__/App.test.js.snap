// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot from App 1`] = `
<DocumentFragment>
  <div
    class="App css-1dqid6c"
  >
    <header
      class="css-10dantf"
    >
      <h1
        class="css-aknc9a"
      >
        TravelBar
      </h1>
      <div
        class="css-p5lzhc"
      >
        <button
          class="css-1t8a2ny"
          data-testid="btnAlcohol"
        >
          Alcoholic
        </button>
        <button
          class="css-1t8a2ny"
          data-testid="btnNoAlcohol"
        >
          Non alcoholic
        </button>
      </div>
    </header>
    <div>
      <p>
        oops...something went wrong 
      </p>
      <div>
        <h1
          class="css-nuuxdy"
        >
          Â¿What's TravelBar?
        </h1>
        <div
          class="css-11w4xf7"
        >
          <div
            class="css-qelhqh"
          >
            <p
              class="css-12kbceo"
            >
              Set within a historic 17th century Spanish house in Central Barcelona.
              <br />
              A casual cocktail bar offering kick ass cocktails, beers, wine and bites late into the night. The playful sibling in the family uses all the experience across our history to create expanded versions of familiar serves, alongside some unexpected gems.
              <br />
              A bar where you can dance, play board games, listen to music... We also have karaoke and open mic nights where you can get to know new talents.
              <br />
              Your kids are more than welcomed here. We have cocktails without alcohol just for them.
              <br />
              <br />
              Just come and have fun!!
            </p>
          </div>
          <div
            class="css-1hwkwgf"
          >
            <img
              class="css-qra5nm"
              src="barPhoto.jpg"
            />
          </div>
        </div>
      </div>
      <h1
        class="css-nj4lov"
        data-testid="selection"
      >
        Selection of Cocktails
      </h1>
      <p>
        oops...something went wrong
      </p>
      <button
        class="css-ref9l6"
        data-testid="btn"
      >
        See More
      </button>
    </div>
  </div>
</DocumentFragment>
`;
